examples/montecarlo.jl

```diff
@@ -24,7 +24,8 @@ function montecarlo(mc_eval::Function, mc_monitor::Function, reps, n_returns, po
     end
 
     if mod(pernode,pooled) != 0
-        pooled = div(pernode, commsize-1)
+        pooled = Int64(div(pernode, commsize-1))
```

Comment:
Hi Erik,
Rather than round pernode up, I prefer to warn the user that the number of
MPI ranks should be changed, so that the user will think about how to
spread the work in an optimal way. I agree that the types of the arguments
could be declared. Perhaps that would help users to understand the function
better, too. I can make this change, or feel free to do it yourself, too.
Cheers,
M.

On Wed, Oct 21, 2015 at 4:57 PM, Erik Schnetter <notifications@github.com>
wrote:

> In examples/montecarlo.jl
> <https://github.com/JuliaParallel/MPI.jl/pull/84#discussion_r42634397>:
>
> > @@ -24,7 +24,8 @@ function montecarlo(mc_eval::Function, mc_monitor::Function, reps, n_returns, po
> >      end
> >
> >      if mod(pernode,pooled) != 0
> > -        pooled = div(pernode, commsize-1)
> > +        pooled = Int64(div(pernode, commsize-1))
>
> At this point, pernode is a Float64, since it is obtained via a division /
> .
>
> The code would be cleaner if the calculation of pernode was changed to
> explicitly round up and be integer.
>
> Also, some of the arguments to montecarlo itself should probably be
> restricted to Integer.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JuliaParallel/MPI.jl/pull/84/files#r42634397>.
>